sdk/core/core/src/az_json_builder.c

```diff
@@ -28,6 +31,80 @@ AZ_NODISCARD static az_result az_json_builder_write_str(
   return AZ_OK;
 }
 
+typedef struct {
+  az_span_action write;
+  az_span value;
+  size_t begin;
+  size_t i;
```

Comment:
I personally don't like the name `i` in this context. `pos`, `position`, `idx`, `index` is better, IMO.

Reply:
And also given that `begin` is not a pointer, something makes me want to see it named as `begin_something`, such as `begin_pos`, or at least `start`

Reply:
I will keep it as is for now because it's an internal structure.